ArduSub/commands_logic.cpp

```diff
@@ -787,11 +787,7 @@ bool Sub::verify_yaw()
     }
 
     // check if we are within 2 degrees of the target heading
-    if (labs(wrap_180_cd(ahrs.yaw_sensor-yaw_look_at_heading)) <= 200) {
-        return true;
-    } else {
-        return false;
-    }
+    return (fabsf(wrap_180_cd(ahrs.yaw_sensor-yaw_look_at_heading)) <= 200);
```

Comment:
Wait, why fabsf here? Both variables are int32 so it will return an int32 right? The wrap will also return an int/long in that case.

Reply:
That's not what clang says here.
```
../../ArduSub/commands_logic.cpp:790:9: warning: using integer absolute value function 'labs' when argument is of floating point type [-Wabsolute-value]
    if (labs(wrap_180_cd(ahrs.yaw_sensor-yaw_look_at_heading)) <= 200) {
        ^
../../ArduSub/commands_logic.cpp:790:9: note: use function 'std::abs' instead
    if (labs(wrap_180_cd(ahrs.yaw_sensor-yaw_look_at_heading)) <= 200) {
        ^~~~
        std::abs
1 warning generated.
```

Reply:
This is strange, I want to understand why. What board is this for?

Reply:
@OXINARF https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Math/AP_Math.h#L108

Reply:
Ok, I figured out the reason, `wrap_180` doesn't have an instance for long (look in AP_Math.cpp). I think I'll change all instances to use intX_t, that's what we use in code anyway.

Reply:
Nevermind what I said above, it's actually because wrap_ methods always return float. That's a bit strange.

Reply:
This should be two commits, one for the logic, one for the fabsf